submissions/f3d0t/friends-app/app.js

```diff
@@ -0,0 +1,202 @@
+	}
+	resetFilters() {
+		this.currentCards = [...this.allCards];
+		this.renderCards();
+	}
+	filterByGender(gender) {
+		if (gender == "male") {
+			this.currentCards = this.allCards.filter(
+				(friendCard) => friendCard["gender"] == "male"
+			);
+		}
+		if (gender == "female") {
+			this.currentCards = this.allCards.filter(
+				(friendCard) => friendCard["gender"] == "female"
+			);
+		}
+		if (gender == "all") {
+			this.currentCards = [...this.allCards];
+		}
+		this.renderCards();
+	}
+	filterBySearch(searchValue) {
+		const filteredArray = this.currentCards.filter(
+			(friendCard) =>
+				Object.values(friendCard)
+					.slice(1)
+					.join(" ")
+					.toLowerCase()
+					.indexOf(searchValue.toLowerCase()) > -1
+		);
+		this.renderCards(filteredArray);
+	}
+	sortByName(sortType) {
+		if (sortType == "az") {
+			this.currentCards = sortObjectsByPropertyValue(
+				this.currentCards,
+				"name",
+				"ascending"
+			);
+		}
+		if (sortType == "za") {
+			this.currentCards = sortObjectsByPropertyValue(
+				this.currentCards,
+				"name",
+				"descending"
+			);
+		}
+		this.renderCards();
+	}
+	sortByAge(sortType) {
+		if (sortType == "ascending") {
+			this.currentCards = sortObjectsByPropertyValue(
+				this.currentCards,
+				"age",
+				"ascending"
+			);
+		}
+		if (sortType == "descending") {
+			this.currentCards = sortObjectsByPropertyValue(
+				this.currentCards,
+				"age",
+				"descending"
+			);
+		}
+		this.renderCards();
+	}
+}
+
+const sortObjectsByPropertyValue = (arrayOfObjects, key, sortOrder) => {
+	const array = arrayOfObjects;
+	const modifier = sortOrder == "ascending" ? 1 : -1;
+	return array.sort((a, b) => {
+		if (a[key] > b[key]) return 1 * modifier;
+		else if (a[key] < b[key]) return -1 * modifier;
+		else return 0;
+	});
+};
+
+class FriendCard {
+	constructor(friendData) {
+		this.gender = friendData.gender;
+		this.name = `${friendData.name.first} ${friendData.name.last}`;
+		this.age = friendData.dob.age;
+		this.photo = friendData.picture.large;
+		this.email = friendData.email;
+		this.phone = friendData.cell;
+		this.country = friendData.location.country;
+	}
+
+	createCard() {
+		const card = document.createElement("div");
+		card.classList.add("card");
+		card.innerHTML = `<img class="card__img" src="${this.photo}" alt="${
+			this.name
+		} photo"'>\n
+                          <h3 class="card__name">${this.name}</h3>\n
+                          <span class="card__age">Age: ${this.age}</span>\n
+                          <a class="card__phone" href="tel:+${this.phone.replace(
+														/\D/g,
+														""
+													)}">${this.phone}</a>\n
+                          <a class="card__email" href="mailto:${this.email}">${
+			this.email
+		}</a>\n
+                          <span class="card__country">${this.country}</span>`;
+		return card;
+	}
+}
+
+const dayNightChange = () => {
+	BODY.classList.toggle("light");
+	DAY_NIGHT.classList.toggle("light");
```

Comment:
`DAY_NIGHT` inside `body`, right? :) Why you need separate class on it?

Reply:
just to avoid weird CSS selectors, fixed that